### DISCUSSION TITLE
[Question] how are the various pb2.py files generated and when?

### DISCUSSION THREAD

**Discussion by simkimsia** [google/mesop#501] [Q&A]
I understand that Bazel is being used.

After I run pip install, the pb2.py files are generated from the .proto files. 

My question is where is the place that trigger the bazel to build and generate the various pb2.py files in the components folder?

**Comment by wwwillchen** [comment]
The way Bazel works is that it forms a dependency graph so you can look up which packages are depending on the python proto targets:
https://github.com/search?q=repo%3Agoogle%2Fmesop%20ui_py_pb2&type=code

**Reply by simkimsia** [reply]
Sorry I still need a bit of clarification

Does it mean that somehow when I run pip install mesop it will trigger bazel to generate the various pb2 files?

Or the pb2 files are generated during the hot reload?

**Reply by wwwillchen** [reply]
No, pip install mesop doesn't trigger bazel. Rather, we use bazel to compile the Mesop framework into a Python package and that's what you're installing with pip. All the proto files are generated ahead of time.
